checker/src/abstract_value.rs

```diff
@@ -2828,14 +2941,62 @@ impl AbstractValueTrait for Rc<AbstractValue> {
         self.get_cached_interval()
     }
 
+    #[logfn_inputs(TRACE)]
+    fn get_cached_octagon(&self) -> Rc<OctagonDomain> {
+        {
+            let mut cached_octagon = self.octagon.borrow_mut();
+            let octagon_opt = cached_octagon.as_ref();
+            if let Some(octagon) = octagon_opt {
+                return octagon.clone();
+            }
+            let octagon = self.get_as_octagon();
+            *cached_octagon = Some(Rc::new(octagon));
+        }
+        self.get_cached_octagon()
+    }
+
+    /// Constructs an element of the Octagon domain for simple expressions.
+    #[logfn_inputs(TRACE)]
+    fn get_as_octagon(&self) -> OctagonDomain {
+        if self.expression_size > k_limits::MAX_EXPRESSION_SIZE / 10 {
+            return octagon_domain::BOTTOM;
+        }
+        match &self.expression {
+            Expression::Top => octagon_domain::TOP,
+            Expression::CompileTimeConstant(ConstantDomain::I128(val)) => (*val).into(),
+            Expression::CompileTimeConstant(ConstantDomain::U128(val)) => (*val).into(),
+            Expression::Add { left, right } => left.get_as_octagon().add(&right.get_as_octagon()),
+            Expression::ConditionalExpression {
+                consequent,
+                alternate,
+                ..
+            } => consequent
+                .get_as_octagon()
+                .widen(&alternate.get_as_octagon()),
+            Expression::Join { left, right, .. } => {
+                left.get_as_octagon().widen(&right.get_as_octagon())
+            }
+            Expression::Mul { left, right } => left.get_as_octagon().mul(&right.get_as_octagon()),
+            Expression::Neg { operand } => operand.get_as_octagon().neg(),
+            Expression::Sub { left, right } => left.get_as_octagon().sub(&right.get_as_octagon()),
+            Expression::Switch { cases, default, .. } => cases
+                .iter()
+                .fold(default.get_as_octagon(), |acc, (_, result)| {
+                    acc.widen(&result.get_as_octagon())
+                }),
+            Expression::Variable { .. } => octagon_domain::TOP,
```

Comment:
But I am not sure about this one. In this case, we don't know much and the best what we can do is to provide TOP.
Is that right?